static/js/composebox_typeahead.js

```diff
@@ -342,6 +343,40 @@ exports.tokenize_compose_str = function (s) {
     return '';
 };
 
+function get_mention_candidates_data() {
+    var all_items = _.map(['all', 'everyone', 'stream'], function (mention) {
+        return {
+            special_item_text: i18n.t("__wildcard_mention_token__ (Notify stream)",
+                                      {wildcard_mention_token: mention}),
+            email: mention,
+            // Always sort above, under the assumption that names will
+            // be longer and only contain "all" as a substring.
+            pm_recipient_count: Infinity,
+            full_name: mention,
+        };
+    });
+    var persons = people.get_realm_persons();
+    var groups = user_groups.get_realm_user_groups();
+    return [].concat(persons, all_items, groups);
+}
```

Comment:
Can you split the extraction of each these functions their own preliminary commit (and maybe more for any reorganizations that aren't part of implementing the new feature), that doesn't make any other changes?  

That would make it much easier to see what the actual logic changes are in the PR.

Reply:
Okay, sure, I'll be splitting these into separate commits. Thanks!